runtime/js/99_main.js

```diff
@@ -67,6 +67,7 @@ import {
   windowOrWorkerGlobalScope,
   workerRuntimeGlobalProperties,
 } from "ext:runtime/98_global_scope.js";
+import * as testing from "ext:cli/40_testing.js";
```

Comment:
This is a red flag - code in `runtime/` shouldn't depend on code from `cli/`. I think you can move `core.setMacrotaskCallback` to `40_testing.js`